File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/dto/UserA11yProfileResponse.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserA11yProfile;

import java.time.LocalDateTime;
import java.util.UUID;

public record UserA11yProfileResponse(
        UUID profileId,
        String profileName,
        String description,
        Integer contrastLevel,
        Integer textSizeLevel,
        Integer textSpacingLevel,
        Integer lineHeightLevel,
        String textAlign,
        Boolean screenReader,
        Boolean smartContrast,
        Boolean highlightLinks,
        Boolean cursorHighlight,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    public static UserA11yProfileResponse fromEntity(UserA11yProfile p) {
        var pi = p.getProfileInfo();
        return new UserA11yProfileResponse(
                p.getProfileId(),
                pi.getProfileName(),
                pi.getDescription(),
                pi.getContrastLevel(),
                pi.getTextSizeLevel(),
                pi.getTextSpacingLevel(),
                pi.getLineHeightLevel(),
                pi.getTextAlign(),
                pi.getScreenReader(),
                pi.getSmartContrast(),
                pi.getHighlightLinks(),
                pi.getCursorHighlight(),
                p.getCreatedAt(),
                p.getUpdatedAt()
        );
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/dto/UserA11yProfileResponse.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserA11yProfile
import java.time.LocalDateTime
import java.util.*

data class UserA11yProfileResponse(
    val profileId: UUID,
    val profileName: String,
    val description: String,
    val contrastLevel: Int,
    val textSizeLevel: Int,
    val textSpacingLevel: Int,
    val lineHeightLevel: Int,
    val textAlign: String,
    val screenReader: Boolean,
    val smartContrast: Boolean,
    val highlightLinks: Boolean,
    val cursorHighlight: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
) {
    companion object {
        fun fromEntity(p: UserA11yProfile): UserA11yProfileResponse {
            val pi = p.getProfileInfo()
            return UserA11yProfileResponse(
                p.profileId,
                pi.profileName,
                pi.description,
                pi.contrastLevel,
                pi.textSizeLevel,
                pi.textSpacingLevel,
                pi.lineHeightLevel,
                pi.textAlign,
                pi.screenReader,
                pi.smartContrast,
                pi.highlightLinks,
                pi.cursorHighlight,
                p.createdAt,
                p.updatedAt
            )
        }
    }
}
